enhancements/storage/csi-ebs-operator.md

```diff
@@ -0,0 +1,142 @@
+---
+title: csi-ebs-operator
+authors:
+  - "@fbertina"
+reviewers:
+  - "@jsafrane”
+  - "@hekumar"
+  - "@chuffman"
+approvers:
+  - "@..."
+creation-date: 2020-03-09
+last-updated: 2020-03-09
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# CSI Operator for AWS EVS
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the past storage drivers have been provided either in-tree or via external provisioners. With the move to advocating CSI as the way to deliver storage drivers, Red Hat needs to move the in-tree drivers (and others that our customers need) to use CSI. With the focus on operators in Red Hat OCP 4.x these CSI drivers should be installed in the form of an Operator.
+
+## Motivation
+
+* In-tree drivers will be removed from Kubernetes, we need to continue to make the drivers available and CSI is the way to do this.
+* AWS is a key cloud provider for OpenShift and use of EBS has been supported in both OpenShift 3.x and 4.x, we need to move this driver to be CSI, provided by an operator.
+* CSI drivers (incl. AWS EBS) will enable new storage features, not possible with in-tree volume plugins, like snapshots and volume cloning.
+
+### Goals
+
+* Create an operator to install the AWS EBS CSI driver.
+* Publish the operator on OperatorHub.
+* Package and ship downstream image of AWS EBS CSI driver (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+### Non-Goals
+
+* Driver creation (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+## Proposal
+
+1. OCP ships a new operator called aws-ebs-csi-driver-operator. This operator:
+   * Installs the AWS EBS CSI driver.
```

Comment:
It should also manage AWS credentials - either through cloud-credential-operator or manually provided by the user.

Reply:
It is probably fine to not have field for AWS credentials in 4.5?

Reply:
I added an item to use `aws-creds` for now.

From `cloud-credential-operator`'s README:
```
Add CredentialsRequests objects to your CVO manifests and deployed via the release payload. Please do not create them in operator code as we want to use the release manifest for auditing and dynamically checking permissions.
```

If we decide to create the credentials ourselves, it'll probably have to be done via OLM.